File: frontend/src/app/dtos/RouteWithFacilitiesDto.ts
export interface GeoJsonPosition {
  latitude: number;
  longitude: number;
  altitude?: number | null;
}

export interface RouteWithFacilityDefaults {
  originalRoute: GeoJsonPosition[];
  includeToilets: boolean;
  toiletIntervalMeters: number;
  includeFountains: boolean;
  fountainIntervalMeters: number;
  maxFacilityDistance: number;
}

File: backend/src/main/java/com/smartroute/smartroute1/endpoint/dto/RouteWithFacilitiesDto.java
package com.smartroute.smartroute1.endpoint.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class RouteWithFacilitiesDto {
    @NotNull
    private String originalRoute;

    private boolean includeToilets = false;

    @Min(1)
    private int toiletIntervalMeters = 5000;

    private boolean includeFountains = false;

    @Min(1)
    private int fountainIntervalMeters = 3000;

    @Min(1)
    private int maxFacilityDistance = 500;
}
